ultravox/inference/infer.py

```diff
@@ -38,6 +38,29 @@ def __init__(
             None
         )
 
+        assert self.tokenizer.padding_side == "left"
+
+    def batch_infer(
+        self,
+        samples: List[datasets.VoiceSample],
+        max_tokens: Optional[int] = None,
+        temperature: Optional[float] = None,
+    ) -> base.InferenceGenerator:
```

Comment:
If the input is a list of `VoiceSample`, why not return a list of `VoiceOutput` as output? It seems conceptually more straightforward.